File: mercado_backend/src/main/java/com/mercado/pagos/servicios/service/DeudaService.java
package com.mercado.pagos.servicios.service;

import com.mercado.pagos.servicios.dto.request.DeudaRequestDTO;
import com.mercado.pagos.servicios.dto.response.PageResponseDTO;
import com.mercado.pagos.servicios.dto.response.DeudaResponseDTO;

import java.time.LocalDateTime;

public interface DeudaService {

    DeudaResponseDTO crearDeuda(DeudaRequestDTO requestDTO);

    PageResponseDTO<DeudaResponseDTO> listarDeudas(int page, int size);

    PageResponseDTO<DeudaResponseDTO> filtrarDeudas(
            String codigoPuesto,
            String estado,
            LocalDateTime inicio,
            LocalDateTime fin,
            int page,
            int size
    );

    DeudaResponseDTO exonerarDeuda(Long id, String motivo);

}

File: mercado_frontend/src/app/features/deudas/services/deuda.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from '../../../core/services/api.service';
import { DeudaRequestDTO, DeudaResponseDTO, EstadoDeuda, PageResponseDTO } from '../models/deuda.models';

@Injectable({
  providedIn: 'root'
})
export class DeudaService {
  private readonly api = inject(ApiService);
  private readonly basePath = '/deudas';

  crear(request: DeudaRequestDTO): Observable<DeudaResponseDTO> {
    return this.api.post<DeudaResponseDTO, DeudaRequestDTO>(this.basePath, request);
  }

  listar(
    page: number,
    size: number,
    filtros?: {
      codigoPuesto?: string;
      estado?: EstadoDeuda | '';
      inicio?: string;
      fin?: string;
    }
  ): Observable<PageResponseDTO<DeudaResponseDTO>> {
    return this.api.get<PageResponseDTO<DeudaResponseDTO>>(this.basePath, {
      page,
      size,
      codigoPuesto: filtros?.codigoPuesto,
      estado: filtros?.estado,
      inicio: filtros?.inicio,
      fin: filtros?.fin
    });
  }

  exonerar(id: number, motivo: string): Observable<DeudaResponseDTO> {
    return this.api.patch<DeudaResponseDTO, null>(
      `${this.basePath}/${id}/exonerar?motivo=${encodeURIComponent(motivo)}`,
      null
    );
  }
}
